### ISSUE TITLE
hot reload hangs on forever with tests with `flutter run`

### ISSUE THREAD

**Issue by nt4f04uNd** [flutter/flutter#79150]
run a test with `flutter run`

hot restart works as intended - it reruns all tests
hot reload hangs on forever

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    await tester.pumpWidget(MyApp());
  });
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Code Sample',
      home: Container(),
    );
  }
}
```
</details>

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 2.1.0-13.0.pre.223, on Microsoft Windows [Version 10.0.19041.867], locale ru-RU)
    • Flutter version 2.1.0-13.0.pre.223 at c:\dev\src\flutter
    • Framework revision 891511d58f (3 days ago), 2021-03-23 14:57:01 -0700
    • Engine revision d59a01d2d3
    • Dart version 2.13.0 (build 2.13.0-150.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\danya\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.54.3)
    • VS Code at C:\Users\danya\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.20.0

[✓] Connected device (4 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)       • windows       • windows-x64    • Microsoft Windows [Version 10.0.19041.867]
    • Chrome (web)            • chrome        • web-javascript • Google Chrome 89.0.4389.90
    • Edge (web)              • edge          • web-javascript • Microsoft Edge 89.0.774.57

• No issues found!
```
</details>

**Comment by darshankawar** [comment]
<details>
<summary>console log</summary>

```
Running pod install...                                           1,062ms
Running Xcode build...                                                  
 └─Compiling, linking and signing...                         5.0s
Xcode build done.                                            9.6s
fopen failed for data file: errno = 2 (No such file or directory)       
Errors found! Invalidating cache...                                     
flutter: 00:00 +0: Counter increments smoke test                        
[VERBOSE-2:FlutterObservatoryPublisher.mm(138)] Could not register as server for FlutterObservatoryPublisher, permission denied. Check your 'Local Network' permissions for this app in the Privacy section of the system Settings.
fopen failed for data file: errno = 2 (No such file or directory)       
Errors found! Invalidating cache...                                     
Installing and launching...                                        23.3s
[VERBOSE-2:FlutterObservatoryPublisher.mm(138)] Could not register as server for FlutterObservatoryPublisher, permission denied. Check your 'Local Network' permissions for this app in the Privacy section of the system Settings.
flutter: 00:01 +1: All tests passed!
Syncing files to device Darshan's iphone...                         56ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on Darshan's iphone is available at: http://127.0.0.1:50808/2Zjr1s0HzOw=/

Flutter DevTools, a Flutter debugger and profiler, on Darshan's iphone is available at: http://127.0.0.1:9102?uri=http%3A%2F%2F127.0.0.1%3A50808%2F2Zjr1s0HzOw%3D%2F

Running with unsound null safety
For more information see https://dart.dev/null-safety/unsound-null-safety

Performing hot restart...                                               
Restarted application in 980ms.
flutter: 00:00 +0: Counter increments smoke test
flutter: 00:00 +1: All tests passed!

Performing hot reload...                                              

```

</details>

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.0.3, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.0.3 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 4d7946a68d (10 days ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.54.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 14.4.1
    • macOS (desktop)           • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 89.0.4389.90

! Doctor found issues in 1 category.

```

</details>

**Comment by jonahwilliams** [comment]
I can repro this but I'm not sure if its something we'll be able to fix
